doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,481 @@
+    visual_tools_.prompt("Press 'next' in the window to continue");
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Publishes the given path without removing old content.
+  */
+  void publishPath(const moveit::planning_interface::MoveGroupInterface::Plan& plan)
+  {
+    // To publish the path
+    visual_tools_.publishTrajectoryLine(plan.trajectory_, joint_model_group_);
+    visual_tools_.trigger();
+    visual_tools_.prompt("Press 'next' in the window to continue");
+  }
+
+  /**
+  * Prints off basic information about move_group_
+  **/
+  void printBasicInfo()
+  {
+    // BEGIN_SUB_TUTORIAL basic_info
+    //
+    // Getting Basic Information
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // move_group holds a lot of information about the robot.
+    //
+    // For example we can print the name of the reference frame for this robot.
+    ROS_INFO_NAMED(name_, "Planning frame: %s", move_group_.getPlanningFrame().c_str());
+
+    // We can also print the name of the end-effector link for this group.
+    ROS_INFO_NAMED(name_, "End effector link: %s", move_group_.getEndEffectorLink().c_str());
+
+    // We even can get a list of all the groups in the robot.
+    ROS_INFO_NAMED(name_, "Available Planning Groups:");
+    std::copy(move_group_.getJointModelGroupNames().begin(), move_group_.getJointModelGroupNames().end(),
+              std::ostream_iterator<std::string>(std::cout, ", "));
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Populates the plan with a path to the given joint space goal
+  * Returns the success of planning
+  **/
+  bool planJointSpaceGoal( const std::vector<double> joint_group_positions, moveit::planning_interface::MoveGroupInterface::Plan& plan)
+  {
+    // Under some circumstances the robot may be unable to move as the plan's start state
+    // isn't where the actual robot is. These can be used to update this
+    /* move_group_.setStartState(*move_group_.getCurrentState()); */
+    /* start_state.setFromIK(joint_model_group_); */
+
+    // BEGIN_SUB_TUTORIAL plan_joint_space
+    // To plan to the goal we must set move_group's target to our goal
+    move_group_.setJointValueTarget(joint_group_positions);
+    // move_group can now plan to this goal
+    bool success = (move_group_.plan(plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    // END_SUB_TUTORIAL'
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan (joint space goal) %s", success ? "" : "FAILED");
+    return success;
+  }
+
+  /**
+  * Populates the plan with a path to the pose goal
+  * Returns the success of planning
+  **/
+  bool planPoseGoal(geometry_msgs::Pose pose, moveit::planning_interface::MoveGroupInterface::Plan& plan, bool resetStartState=false)
+  {
+    // Someitmes it can be useful to plan a path that doesn't start where the current robot is. If that is done it is important
+    // to reset the planning start state back to where the robot is when done. This is also used when an object is attached.
+    if (resetStartState)
```

Comment:
Probably minor, but the default behavior should be to use the current state. 
Better would be to optionally pass in a start state and if it is not null then setStartState. In this approach you will have to clear the start state at the end.

Reply:
When you say in this approach, do you mean in the current approach or in your proposed new method?

Reply:
proposed new approach